core/src/test/java/org/elasticsearch/env/EnvironmentTests.java

```diff
@@ -71,4 +80,53 @@ public void testRepositoryResolution() throws IOException {
         assertThat(environment.resolveRepoURL(new URL("jar:http://localhost/test/../repo1?blah!/repo/")), nullValue());
     }
 
+    public void testDefaultPathDataSet() {
+        final Settings settings = Settings.builder()
+                .put("path.home", "/home")
+                .put("path.data.0", "/mnt/zero")
+                .put("path.data.1", "/mnt/one")
+                .put("default.path.data", "/mnt/default")
+                .build();
+        final Environment environment = new Environment(settings);
+        final Set<Path> dataFiles = new HashSet<>(Arrays.asList(environment.dataFiles()));
+        assertThat(dataFiles, equalTo(new HashSet<>(Arrays.asList(PathUtils.get("/mnt/zero"), PathUtils.get("/mnt/one")))));
+        assertThat(environment.defaultPathData(), equalTo(PathUtils.get("/mnt/default")));
```

Comment:
Can you add one of those `Version.current.lessThanOrEqualTo(Version.fromId(<whatever ID makes 7.0.0-alpha1>)` things so we can know we can drop this then?